docs/_posts/2021-07-05-whats-next.md

```diff
@@ -0,0 +1,176 @@
+
+Currently we do not offer bindings in `lib-state` for saving and restoring state to survive activity or process recreation. This is something we could add (based on [`Saver`](https://developer.android.com/reference/kotlin/androidx/compose/runtime/saveable/Saver)), specifically for scoped stores.
+
+```kotlin
+val store = scopedSaveableStore<BrowserScreenState, BrowserScreenAction> { restoredState ->
+    BrowserScreenStore(restoredState ?: BrowserScreenState())
+}
+```
+
+### App State
+
+_"App state"_ is the state the app is in, independently from the currently displayed screen (e.g. “Is the app in light or dark mode?”).
+
+Currently in Fenix there’s no centralized app state. For some parts there are manager singletons (e.g. `ThemeManager`) or the state is read from `SharedPreferences`.
+
+We could try using an global app store and the same patterns we use for the browser store and the screen scoped stores. That’s something we are currently trying in Focus ([AppStore](https://github.com/mozilla-mobile/focus-android/blob/main/app/src/main/java/org/mozilla/focus/state/AppStore.kt)). In fact, in Focus, the screen scoped state is a sub state of the application-wide state ([Screen state](https://github.com/mozilla-mobile/focus-android/blob/main/app/src/main/java/org/mozilla/focus/state/AppState.kt#L26)).
+
+### Stateless composables
+
+With Jetpack Compose it is preferred to write stateless composables ([State hoisting](https://developer.android.com/jetpack/compose/state#state-hoisting)). This means that state is passed down as function parameters and events are passed up by invoking functions. Listening to the store and dispatching actions sidesteps this mechanism. Only subscribing to state changes at the top layer and passing everything down/up is cumbersome and may introduce a lot of duplicated “glue” code across our projects.
+
+Let’s look at a simplified example of a browser toolbar. There are two options:
+
+**Example A**: All state (e.g. the URL to display) gets passed down to the toolbar:
+
+```Kotlin
+@Composable
+fun Toolbar(
+    url: String
+) {
+    Text(url)
+}
+```
+
+**Example B**: The toolbar subscribes to state it needs itself.
+
+```Kotlin
+@Composable
+fun Toolbar(
+    store: BrowserStore
+) {
+    val url = store.observeAsState { state -> state.selectedTab?.content?.url }
+    Text(url.value ?: "")
+}
+```
+
+The code from example **A** is the most reusable. The app is in full control of what state gets displayed. But this also introduces duplicate code across apps (for getting the state and passing it down) and makes it more likely to introduce bugs (security and spoofing). Example **B** is guaranteed to be consistent across apps. But the composable is strictly tied to the state and how it gets observed.
```

Comment:
When prototyping a BrowserTabsTray, I ran into this quandry and ended up writing two versions. 😄 

Using a shim to connect an existing API saved me from writing duplicate code that would need to handle observing private and normal tabs, which is what observing the BrowserStore directly would force me to do.

Reply:
"Private" vs. "Normal" tabs is an interesting use case. Maybe a helper like the `Target` class could be helpful here.